### ISSUE TITLE
Question: why wrap in a span element?

### ISSUE THREAD

**Issue by inspector71** [alphagov/accessible-autocomplete#402]
Hello

Thank you for this code. Clearly a lot effort has gone into it.

Hope it's ok to ask here (I have read as much documentation as I can find) but is there any particular accessibility or other reason why, at least when using the `.enhanceSelectElement()` wrapper, the `autocomplete__wrapper` `<div>` itself is wrapped in a `<span>` element?

Regards

**Comment by nickcolley** [comment]
Heya @inspector71,

Had a look into this, the element is used to mount the new component alongside the `select` element.

There's no impact on accessibility however I've changed it from a `span` to a `div` as this is more correct.

See the pull request for more details: https://github.com/alphagov/accessible-autocomplete/pull/407

**Comment by inspector71** [comment]
Hi @nickcolley 

Thanks for your input! Great to see activity in this repo. 

I guess my question was two-fold.  Why the `span` and why is a 'wrapper' element ... outside an existing 'wrapper' element :)

I realise it's not unknown for code to use _multiple_ wrapper elements, but it's also possibly a bit messy?

In my case, though I am not suggesting this is a common pattern (though it may be for all I know), I wanted to use `display:grid` to align a `<label>` field alongside it's relevant `<select>` field. Using `enhanceSelectElement()` the addition of the `span` wrapper broke that layout. I could nest the `<select>` and or `<label>` fields, as I'm sure a lot of people do, but I was hoping to avoid that extra 'overhead'.

I started tinkering on a codepen for example purposes but came up against a wierdo bug with my system. Would a codepen help anyone? If it might, I'll put in the time.

**Comment by nickcolley** [comment]
@inspector71 the top level 'wrapper' element is not necessary in this specific example but is there for consistency with the other examples.

> it's also possibly a bit messy

You are not required to use this top level 'wrapper' as seen in the examples in your own code.

See the progressive enhancement documentation for details: https://github.com/alphagov/accessible-autocomplete#progressive-enhancement

**Comment by inspector71** [comment]
> 
> 
> @inspector71 the top level 'wrapper' element is not necessary in this specific example but is there for consistency with the other examples.
> 
> > it's also possibly a bit messy
> 
> You are not required to use this top level 'wrapper' as seen in the examples in your own code.
> 
> See the progressive enhancement documentation for details: https://github.com/alphagov/accessible-autocomplete#progressive-enhancement

Thanks @nickcolley but I suspect you may be incorrect.

I did use that documentation and my original question was based on the resulting generated code.

For me, this part of the documentation says it all:

<blockquote>
<p><strong>Note</strong>: The <code>accessibleAutocomplete.enhanceSelectElement</code> function is fairly light and wraps the public API for <code>accessibleAutocomplete</code>. If your use case doesn't fit the above defaults, try <a href="/alphagov/accessible-autocomplete/blob/master/src/wrapper.js">reading the source</a> and seeing if you can write your own.</p>
</blockquote>

In other words, we threw something together to work with existing `select` elements but it's a low-fi or minimal effort. If you need better try writing it yourself.

Unfortunately I am not able to invest the time to do that. A11y is also not a high requirement as I am only targeting two users, one of whom is myself.

Best of luck with the project everyone.
